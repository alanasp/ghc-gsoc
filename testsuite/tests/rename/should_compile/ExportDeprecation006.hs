{-# OPTIONS_GHC -fwarn-warnings-deprecations #-}

-- Test deprecation of variable and data constructor exports

module ShouldCompile where

import ExportDeprecation006_B(func_C) --deprecated
import ExportDeprecation006_C(func_C) --deprecated at definition
import ExportDeprecation006_A(func_C) --deprecated

main = func_C --warning because deprecated at the defining module
