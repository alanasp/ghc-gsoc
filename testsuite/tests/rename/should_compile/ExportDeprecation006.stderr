ExportDeprecation006.hs:7:1: warning: [-Wdeprecations (in -Wdefault)]
    Deprecated: "ExportDeprecation006_C shall be imported directly"

ExportDeprecation006.hs:8:1: warning: [-Wdeprecations (in -Wdefault)]
    Deprecated: "func_C is deprecated"

ExportDeprecation006.hs:9:1: warning: [-Wdeprecations (in -Wdefault)]
    Deprecated: "ExportDeprecation006_C shall be imported directly"

ExportDeprecation006.hs:11:8: warning: [-Wdeprecations (in -Wdefault)]
    In the use of ‘func_C’
    (imported from ExportDeprecation006_A, but defined in ExportDeprecation006_C):
    Deprecated: "ExportDeprecation006_C shall be imported directly"
